R/filter2.R

```diff
@@ -45,10 +45,21 @@ filter2 = function(x, filter, boundary = c("circular", "replicate")) {
          },
          ## pad with a given value
          linear = {
-           dx[1:2] = dx[1:2] + cf[1:2]
-           xpad = array(val, dx)
-           ## is there a better way of doing this?
-           x = do.call("[<-", c(quote(xpad), lapply(d, function(x) enquote(1:x)), quote(x)) )
+           x = imageData(x)
+           
+           dx[1:2] = dx[1:2] + df[1:2] - 1L
```

Comment:
Actually, my point was that the padding for `boundary = "linear"` doesn't need to be as wide as for the `"replicate"` mode. The reason is that in the former case the padding value is uniform around the image, therefore the same padding pixels can be reused when filtering at the left image edge, as when processing the right one (this is possible because the image is wrapped around the borders). You can verify that this indeed works by comparing the results from your code to my original implementation.

If you don't mind, I would prefer to keep the previous minimal padding, i.e. have
```r
dx[1:2] = dx[1:2] + cf[1:2]
```